packages/webapp/pages/earnings.tsx

```diff
@@ -0,0 +1,408 @@
+import type { UserImageProps } from '@dailydotdev/shared/src/components/ProfilePicture';
+import {
+  ProfileImageSize,
+  ProfilePicture,
+} from '@dailydotdev/shared/src/components/ProfilePicture';
+import { TimeFormatType } from '@dailydotdev/shared/src/lib/dateFormat';
+import { Separator } from '@dailydotdev/shared/src/components/cards/common/common';
+import { getLayout as getFooterNavBarLayout } from '../components/layouts/FooterNavBarLayout';
+import { getLayout } from '../components/layouts/MainLayout';
+import ProtectedPage from '../components/ProtectedPage';
+
+type BuyCoreProps = {
+  amount: number;
+  price: number;
+};
+const BuyCore = ({ amount, price }: BuyCoreProps): ReactElement => {
+  return (
+    <div className="flex flex-1 flex-col items-center rounded-14 bg-surface-float p-2">
+      <CoinIcon size={IconSize.XLarge} className="mb-1" />
+      <Typography type={TypographyType.Title3} bold>
+        {amount}
+      </Typography>
+      <Typography
+        type={TypographyType.Caption2}
+        color={TypographyColor.Tertiary}
+      >
+        ${price}
+      </Typography>
+    </div>
+  );
+};
+
+type BalanceBlockProps = {
+  Icon: ReactElement;
+  title: string;
+  description: string;
+  balance: number;
+} & WithClassNameProps;
+const BalanceBlock = ({
+  Icon,
+  balance,
+  title,
+  description,
+  className,
+}: BalanceBlockProps): ReactElement => {
+  return (
+    <div
+      className={classNames('flex flex-1 flex-col rounded-14 p-4', className)}
+    >
+      {Icon}
+      <SimpleTooltip content={description} appendTo="parent">
+        <div className="mt-4 flex gap-1">
+          <Typography type={TypographyType.Callout}>{title}</Typography>
+          <InfoIcon />
+        </div>
+      </SimpleTooltip>
+      <Typography type={TypographyType.Title2} bold>
+        {balance}
+      </Typography>
+    </div>
+  );
+};
+
+const Divider = classed('div', 'h-px w-full bg-border-subtlest-tertiary');
+
+type TransactionItemProps = {
+  type: 'tip' | 'awardCredit' | 'awardDebit' | 'purchase';
+  user: UserImageProps;
+  date: Date;
+  amount: number;
+};
+
+const TransactionTypeToIcon: Record<
+  TransactionItemProps['type'],
+  ReactElement
+> = {
+  tip: (
+    <div className="size-4 rounded-10 bg-action-upvote-float text-accent-avocado-default">
+      <PlusIcon size={IconSize.Size16} />
+    </div>
+  ),
+  awardCredit: (
+    <div className="size-4 rounded-10 bg-action-upvote-float text-accent-avocado-default">
+      <PlusIcon size={IconSize.Size16} />
+    </div>
+  ),
+  awardDebit: (
+    <div className="size-4 rounded-10 bg-action-downvote-float text-accent-ketchup-default">
+      <MinusIcon size={IconSize.Size16} />
+    </div>
+  ),
+  purchase: (
+    <div className="size-4 rounded-10 bg-action-bookmark-float text-accent-bun-default">
+      <CoinIcon size={IconSize.Size16} />
+    </div>
+  ),
+};
+const TransactionItem = ({
```

Comment:
Split to sub components probably, but we can also do it when connecting data.

Reply:
Yeah 100% just wasn't sure if these final/re-used yet so for quicker iteration leaving.
Should be extracted ideally, some also are super similar now.